fiftyone/factory/repos/execution_store.py

```diff
@@ -102,53 +128,79 @@ def set_key(self, store_name, key, value, ttl=None) -> KeyDocument:
     def get_key(self, store_name, key) -> KeyDocument:
         """Gets a key from the specified store."""
         raw_key_doc = self._collection.find_one(
-            _where(store_name, key, self._dataset_id)
+            dict(store_name=store_name, key=key, dataset_id=self._dataset_id)
         )
         key_doc = KeyDocument(**raw_key_doc) if raw_key_doc else None
         return key_doc
 
-    def list_keys(self, store_name) -> list[str]:
-        """Lists all keys in the specified store."""
-        keys = self._collection.find(
-            _where(store_name, dataset_id=self._dataset_id), {"key": 1}
-        )
-        return [key["key"] for key in keys]
-
     def update_ttl(self, store_name, key, ttl) -> bool:
         """Updates the TTL for a key."""
         expiration = KeyDocument.get_expiration(ttl)
         result = self._collection.update_one(
-            _where(store_name, key, self._dataset_id),
+            dict(store_name=store_name, key=key, dataset_id=self._dataset_id),
             {"$set": {"expires_at": expiration}},
         )
         return result.modified_count > 0
 
     def delete_key(self, store_name, key) -> bool:
-        """Deletes the document that matches the store name and key."""
+        """Deletes the document that matches the store name and key, if one
+        exists.
+        """
         result = self._collection.delete_one(
-            _where(store_name, key, self._dataset_id)
+            dict(store_name=store_name, key=key, dataset_id=self._dataset_id)
         )
         return result.deleted_count > 0
 
-    def delete_store(self, store_name) -> int:
-        """Deletes the entire store."""
+    def list_keys(self, store_name) -> list[str]:
+        """Lists all keys in the specified store."""
+        result = self._collection.find(
+            dict(store_name=store_name, dataset_id=self._dataset_id),
+            {"key": 1},
+        )
+        return [d["key"] for d in result]
+
+    def count_keys(self, store_name) -> int:
+        """Counts the keys in the specified store."""
+        return self._collection.count_documents(
+            dict(store_name=store_name, dataset_id=self._dataset_id)
+        )
```

Comment:
This is a good suggestion. I also think this could be added to the `_where()` util instead of removing it. That util could also be a method like so:

```python
query = self._where_key(store_name) # handles $ne and self.dataset_id
```

Reply:
I implemented the `{"$ne": "__store__"}` suggestion in https://github.com/voxel51/fiftyone/pull/5086/commits/800cdad9963a8bd3cc15d525513456214e9e8ced.

It felt a bit magic to me to embed this logic in a `_where()` utility for minimal benefit. Just my opinion